src/hypercall.rs

```diff
@@ -144,26 +147,44 @@ pub fn open(mem: &MmapMemory, sysopen: &mut OpenParams, file_map: &mut UhyveFile
 }
 
 /// Handles an close syscall by closing the file on the host.
-pub fn close(sysclose: &mut CloseParams) {
-	unsafe {
-		sysclose.ret = libc::close(sysclose.fd);
+pub fn close(sysclose: &mut CloseParams, file_map: &mut UhyveFileMap) {
+	if file_map.fdmap.is_fd_present(sysclose.fd.into_raw_fd()) {
+		if sysclose.fd > 2 {
+			unsafe { sysclose.ret = libc::close(sysclose.fd) }
+			file_map.fdmap.remove_fd(sysclose.fd)
+		} else {
+			// Ignore closes of stdin, stdout and stderr that would
+			// otherwise affect Uhyve
+			sysclose.ret = 0
```

Comment:
Can this path be triggered now at all?

Reply:
Yes, because the first condition succeeds if sysclose.fd is 0, 1, or 2. It is "present" - as in, it is among the file descriptors that the guest should be able to use - but we don't track it in the internal data structures.